packages/iTwinUI-react/src/core/utils/components/MiddleTextTruncation.tsx

```diff
@@ -52,6 +56,7 @@ export const MiddleTextTruncation = (props: MiddleTextTruncationProps) => {
         ...style,
       }}
       ref={ref}
+      title={hasOverflow ? title : undefined}
```

Comment:
I would say this should be opt-in rather than rule, have some additional param in MiddleTextTruncationProps for that. Also - I see that you created this PR for your fork rather than for the original repository. You should create pull request from your fork feature branch to main repo main branch instead

Reply:
So, I discussed this feature with one of the itwins developers, and he suggested using onTruncate callback instead, to make it more general.

Reply:
And how it will help us not to show tooltip if text is not truncated?